### ISSUE TITLE
Switch to MariaDB

### ISSUE THREAD

**Issue by MilesBHuff** [SettleScape/server#21]
Storing data in a centralized client-server database instead of individual SQLite databases will make database management easier, although it may result in more ongoing maintenance responsibilities.

**Comment by MilesBHuff** [comment]
@krislamo says there may be a performance hit from moving to MySQL.  I'll need to make sure that any such impact is negligible if I go through with this.

**Comment by krislamo** [comment]
In the spirit of free software, I would like to suggest MariaDB over MySQL if you end up going through with this. I don't think this change will matter too much, but you will have to manage another process.

**Comment by MilesBHuff** [comment]
I do like me some free software, but everything that offers a proper database connection explicitly expects MySQL.  Is MariaDB 100% compatible?

**Comment by krislamo** [comment]
[It should be fine](https://mariadb.com/kb/en/mariadb-vs-mysql-compatibility/) in most cases. Just a random suggestion based on personal preferences. I don't know all the details and/or caveats of your requirements.

**Comment by MilesBHuff** [comment]
"Should be fine in most cases" is good, but but not ideal.  Even if it turns out to be fine for our current plugins, that doesn't mean that some future plugin we add will play nicely with MariaDB;  and that could mean a second migration in the future.

**Comment by MilesBHuff** [comment]
Read this in your link:  "We do monthly merges with the MySQL code base to ensure that MariaDB has any relevant bug fixes added to MySQL.".  This is reassuring -- it sounds like it's a fork of MySQL, which might make it okay.  It also sounds like you can switch between the two without really migrating.

**Comment by krislamo** [comment]
Yeah, I get that. I would be **very** surprised if that ever happened in your situation since all MySQL connectors are identical with MariaDB. If you're worried about it just go MySQL. Like I said earlier, I'm just biased towards MariaDB.

**Comment by MilesBHuff** [comment]
One significant downside of using MySQL instead of SQLite files, is that the norm for plugins that support MySQL is to have plaintext passwords inside the main config files.  It would essentially mean we couldn't commit certain files from the server without publicly revealing passwords.

Another downside is developers would have to run their own instances of MySQL in order to test new code locally.  This might be easily done through the `start.sh` script, though.

**Comment by krislamo** [comment]
The first downside could be eliminated with a configuration management system like Ansible. One of these days I may get around to working more on my ansible-minecraft project and propose using it on SettleScape. There will need to be discussion and research on that migration first. I'll open an issue for discussion when/if I get somewhere on that.

**Comment by MilesBHuff** [comment]
Interesting!  I don't know anything about Ansible.  I'm curious to see what you come up with!  :)

**Comment by MilesBHuff** [comment]
Idk, I'm having trouble seeing the benefits of MySQL over SQLite for this.  Closing for now;  can re-open later if need-be.

**Comment by MilesBHuff** [comment]
ChatGPT seems to be very bullish on MariaDB over MySQL for a myriad of reasons.

**Comment by MilesBHuff** [comment]
Having everything in one database would make fulfilling GDPR requests and such more-realistic.
